File: reverseMatrix/JS/matrix.js
const matrix = array => {
let sum = 0;
let forIndex = array.length;
let maxI = forIndex - 1;
for (let i = 0; i < forIndex / 2; i++) {
  for (let j = 0; j < forIndex / 2; j++) {
    sum += Math.max(Math.max(array[i][j], array[maxI - i][j]), 
      Math.max(array[maxI - i][maxI - j], array[i][maxI - j]));
  }
}
return sum;
}

module.exports = matrix;

File: reverseMatrix/Java/Matrix.java
import java.util.List;

public class Matrix {
    public static int flippingMatrix(List<List<Integer>> matrix) {
        int sum = 0;
        int maxI = matrix.size();
        int maxIPossible = maxI - 1;
        int maxJ = matrix.get(0).size();
        int maxJPossible = maxJ -1;

        for (int i = 0; i < maxI /2; i++) {
            for (int j = 0; j < maxJ / 2; j++) {
                sum += Math.max(Math.max(
                                matrix.get(i).get(j), matrix.get(maxIPossible-i).get(j)),
                        Math.max(matrix.get(maxIPossible-i).get(maxJPossible - j),
                                matrix.get(i).get(maxJPossible - j)));
            }
        }
        return sum;
    }
}
